CardView's margins issue fix on thread responses screen

miankhalid opened the pull request:
https://openedx.atlassian.net/browse/MA-1322

`CardView` adds extra padding on pre-lollipop devices for shadows which causes margins to be increased. So, we've set that extra padding to appear on lollipop devices as well and have subtracted the extra amount from margins to get the desired results.

Screen looks like this now:
![add_comment_rounded_bottom_bg](https://cloud.githubusercontent.com/assets/1710804/10670109/6c10c5e2-78fd-11e5-9899-d653404aaac5.png)

@aleffert @bguertin @1zaman plz review

1zaman commented on VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseDiscussionResponsesAdapter.java:
```diff
@@ -84,6 +84,15 @@ public BasePagination getPagination() {
+        params.topMargin -= discussionResponseRow.getPaddingTop();
+        params.leftMargin -= discussionResponseRow.getPaddingLeft();
+        params.rightMargin -= discussionResponseRow.getPaddingRight();
+        discussionResponseRow.setLayoutParams(params);
```
Hmm, should we have a utility method for this?

bguertin commented:
In the screenshot, I can also see that the background for "Add a comment" is not full bleed, there's a little extra padding. That can also be removed by setting `cardPreventCornerOverlap="false"`

miankhalid commented:
@bguertin setting `cardPreventCornerOverlap="false"` messes up the rounded border at the bottom by overlapping "Add a comment" button's background over it.
![corners_not_rounded](https://cloud.githubusercontent.com/assets/1710804/10360248/b06bc408-6db8-11e5-800c-ae03bd3f4bdf.png)

1zaman commented:
> Due to expensive nature of rounded corner clipping, on platforms before L, CardView does not clip its children that intersect with rounded corners. Instead, it adds padding to avoid such intersection (See [`setPreventCornerOverlap(boolean)`](https://developer.android.com/reference/android/support/v7/widget/CardView.html#setPreventCornerOverlap(boolean)) to change this behavior).

So I guess the only way to backport the clipping is to implement the rounded corners in the overlay ourselves manually :frowning:

bguertin commented:
> So I guess the only way to backport the clipping is to implement the rounded corners in the overlay ourselves manually

A simpler option is to just round the courners of the add comment background using a radius closely matching the card's radius. It wouldn't be perfect, but it would be close. :confused:

bguertin commented:
@miankhalid Could you update this branch please? It's a little out-of-date unfortunately.

I still think using a bottom radius on the comment background is a good enough solution to the corner overlap problem.

miankhalid commented:
Will do.

I had stopped working on this PR in favor of more priority issues of the current sprint e.g. Assessments. But I'll fix this today.

miankhalid commented:
As promised, I've updated the branch and fixed the issue as suggested.
Updated screen looks like this: https://github.com/edx/edx-app-android/pull/384#issue-110000008

@bguertin plz review

bguertin commented:
:+1:
